### DISCUSSION TITLE
Is it leaky?

### DISCUSSION THREAD

**Discussion by ricklabanca** [community/community#50622] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I'm wondering who joshuaschleifer is, because that name is leaked into my suggestions and if he minds!

Now I don't know if this comes from open source scanning or this person getting his private code comments pushed out. Is there a way to tell where suggestions were sourced from?

**Comment by Akash1134** [comment, marked as answer]
Hi, @ricklabanca Thanks for helping make GitHub safe for everyone. GitHub takes the security of our software products and services seriously, including all of the open source code repositories managed through our GitHub organizations, such as [GitHub](https://github.com/GitHub). Please report such cases through our [coordinated disclosure](https://github.com/github/copilot-docs/security/policy#reporting-security-issues).

Because Codex, the model powering GitHub Copilot, was trained on publicly available code, its training set included public personal data that was included in that code. From our internal testing, we found it to be very rare that GitHub Copilot suggestions included personal data verbatim from the training set. In some cases, the model will suggest what appears to be personal data – email addresses, phone numbers, etc. – but those suggestions are actually fictitious information synthesized from patterns in training data and therefore do not relate to any particular individual. For example, when one of our engineers prompted GitHub Copilot with, “My name is Mona and my birthdate is,” GitHub Copilot suggested a random, fictitious date of “December 12,” which is not Mona’s actual birthdate. We have also implemented a filter that blocks emails when shown in standard formats, but it’s still possible to get the model to suggest this sort of content if you try hard enough. 

We are and will keep improving the filter system to be more intelligent to detect and remove more personal data from the GitHub Copilot suggestions.

**Reply by ricklabanca** [reply]
The question is would my code or name appear in someone else’s data as a side effect of telemetry training ?

**Reply by Akash1134** [reply]
Hi @ricklabanca It's very rare personal code snippets or details will ever appear in someone's data. We have implemented a filter that blocks emails when shown in standard formats. However, it’s still possible to get the model to suggest this sort of content if you try hard enough like specifically asking for name or birthdate (referring to public repos).
